ckanext/datastore/logic/action.py

```diff
@@ -431,6 +431,36 @@ def datastore_delete(context: Context, data_dict: dict[str, Any]):
     return result
 
 
+def datastore_records_delete(context: Context, data_dict: dict[str, Any]):
+    '''Deletes all records or a set of records from the DataStore.
+
+    :param resource_id: resource id that the data will be deleted from.
+                        (optional)
+    :type resource_id: string
+    :param force: set to True to edit a read-only resource
+    :type force: bool (optional, default: False)
+    :param filters: filters to apply before deleting (eg {"name": "fred"}).
+                   If missing delete whole table and all dependent views.
+                   (optional)
+    :type filters: dictionary
+    :param calculate_record_count: updates the stored count of records, used to
+        optimize datastore_search in combination with the
+        `total_estimation_threshold` parameter. If doing a series of requests
+        to change a resource, you only need to set this to True on the last
+        request.
+    :type calculate_record_count: bool (optional, default: False)
+
+    **Results:**
+
+    :returns: Original filters sent.
+    :rtype: dictionary
+
+    '''
+    filters = data_dict.pop('filters', {})
+    data_dict['filters'] = filters
```

Comment:
```suggestion
    filters = data_dict.get('filters', None)
    # passing filters=None will cause table to be deleted, use {} to remove records instead
    data_dict['filters'] = {} if filters is None else filters
```

Reply:
Updated!

This is in the case that a user calls the action and passes `filters: None`? In such a case, the `data_dict.pop` would not use the default value of an empty dict, because it would exists as `None` already?

Reply:
that's right

```
>>> help(dict.get)
Help on method_descriptor:

get(...)
    D.get(k[,d]) -> D[k] if k in D, else d.  d defaults to None.
```